neuroanalysis/test_pulse.py

```diff
@@ -25,27 +32,72 @@ def __init__(self, rec: PatchClampRecording, indices=None):
             raise ValueError("Found multiple square pulse in command waveform. Consider using `indices`.")
         pulse = pulses[0]
         pulse.description = 'test pulse'
+        stimulus = pulse
+
         super().__init__(
             recording=rec,
             device_type=rec.device_type,
             device_id=rec.device_id,
             start_time=rec.start_time,
-            channels={'primary': pri, 'command': cmd},
-            stimulus=pulse,
+            channels=channels,
+            stimulus=stimulus,
             clamp_mode=rec.clamp_mode,
-            holding_potential=rec._meta['holding_potential'],
-            holding_current=rec._meta['holding_current'],
-            bridge_balance=rec._meta['bridge_balance'],
-            lpf_cutoff=rec._meta['lpf_cutoff'],
-            pipette_offset=rec._meta['pipette_offset'],
+            holding_potential=rec.meta['holding_potential'],
+            holding_current=rec.meta['holding_current'],
+            bridge_balance=rec.meta['bridge_balance'],
+            lpf_cutoff=rec.meta['lpf_cutoff'],
+            pipette_offset=rec.meta['pipette_offset'],
         )
         self._analysis = None
         # expose these for display and debugging
+        self._main_fit_region = None
         self.main_fit_result = None
         self.main_fit_trace = None
         self.fit_result_with_transient = None
         self.fit_trace_with_transient = None
-        self.initial_double_fit_trace = None
+
+    def save(self):
```

Comment:
Seems like this wil make it awkward to add a Recording.save later on.. Could this done more cleanly by starting there instead? And also adding TSeries.save/load?